server/server.go

```diff
@@ -104,6 +104,10 @@ type Config struct {
 	HTTPServerReadHeaderTimeout   time.Duration `yaml:"http_server_read_header_timeout"`
 	HTTPServerWriteTimeout        time.Duration `yaml:"http_server_write_timeout"`
 	HTTPServerIdleTimeout         time.Duration `yaml:"http_server_idle_timeout"`
+	// HTTPServerWriteTimeout has some misleading behavior. Consider using the below fields to
+	// enable a timeout handler that will write a response after the timeout. See tests for details.
+	HTTPServerUseTimeoutHandler     bool   `yaml:"-"`
+	HTTPServerTimeoutHandlerMessage string `yaml:"-"`
```

Comment:
why not make those configurable via CLI too?

Reply:
aha, i see your point. Some more ideas: `HTTPServerGracefulTimeoutsEnabled`, `HTTPServerGracefulTimeoutsMessage`. But your suggestions looks good to me, I'm happy to go with them too. 

I'm not sure about adding a customized message, can't we get away with a hardcoded string `request timeout reached`?